numpy/core/arrayprint.py

```diff
@@ -869,7 +859,14 @@ def __call__(self, arr):
 
 
 class StructureFormat(object):
-    def __init__(self, format_functions):
+    def __init__(self, data, **options):
```

Comment:
Is this too back-incompatible, or can we let the signature change slide? (Ie, is anyone besides us really calling StructureFormat?)

Reply:
Why not just add `StructFormat.from_data`? I think the old signature was handy

Reply:
handy, yet never used, now.

In any case, good idea to avoid any compatibility issue, done.